filecoin-proofs/src/param.rs

```diff
@@ -27,6 +31,20 @@ pub struct ParameterData {
     pub digest: String,
 }
 
+struct FetchProgress<R> {
+    inner: R,
+    progress_bar: ProgressBar<Stdout>,
+}
+
+impl<R: Read> Read for FetchProgress<R> {
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        self.inner.read(buf).map(|n| {
```

Comment:
you can use [`for_each`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.for_each) instead of `map` as you are not modifying the values